### PR TITLE
Use SourceBranch vs. SourceBranchName

### PR DESCRIPTION
**Pull request by mmitche**
SourceBranchName only evaluates to the last path name of the branch:
- 3.0 in release/3.0
We want the full branch, with refs/heads for consistency with the rest of the repos

### CODE REVIEW

**Comment by mmitche** [comment]
I'll cherry-pick this over into release/3.0 after this.

**Comment by dagood** [comment]
FWIW, in core-setup we use this to find the typical branch name:

```yaml
    - powershell: |
        $prefix = "refs/heads/"
        $branch = "$(Build.SourceBranch)"
        $branchName = $branch
        if ($branchName.StartsWith($prefix))
        {
          $branchName = $branchName.Substring($prefix.Length)
        }
        Write-Host "For Build.SourceBranch $branch, FullBranchName is $branchName"
        Write-Host "##vso[task.setvariable variable=FullBranchName;]$branchName"
      displayName: Find true SourceBranchName
```

File: eng/pipelines/publish.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
                 /t:PublishPackagesToBlobFeed
                 /p:ManifestBuildId=$(Build.BuildNumber)
                 /p:ManifestBuildData=Location=$(_dotnetFeedUrl)
-                /p:ManifestBranch=$(Build.SourceBranchName)
+                /p:ManifestBranch=$(Build.SourceBranch)
```
Comment (dagood) [inline eng/pipelines/publish.yml]: dotnet/versions is also affected, updating `build-info/dotnet/corefx/3.0` at the moment. `release/3.0` is normal, there. I'm not sure it needs to be changed, though--it seems unlikely corefx will start producing some other branch that also ends in `/3.0`.

Comment (dagood) [reply eng/pipelines/publish.yml]: The Tizen folks may care about the inconsistent naming, or at least need to know it's different so they can trigger their builds.

Comment (mmitche) [reply eng/pipelines/publish.yml]: I didn't want to mess with it too much.  AFAIK, there's no way to the string.replace to strip away the refs/heads in yml. I think leaving it as 3.0 is probably fine for now.

File: eng/pipelines/publish.yml [hunk 2]
```diff
@@ -32,6 +32,18 @@ jobs:
...
+        $prefix = "refs/heads/"
+        $branch = "$(Build.SourceBranch)"
+        $branchName = $branch
+        if ($branchName.StartsWith($prefix))
+        {
+          $branchName = $branchName.Substring($prefix.Length)
+        }
+        Write-Host "For Build.SourceBranch $branch, FullBranchName is $branchName"
+        Write-Host "##vso[task.setvariable variable=FullBranchName;]$branchName"
+      displayName: Find true SourceBranchName
```
Comment (dagood) [inline eng/pipelines/publish.yml]: The indentation looks wrong here, I think everything after `- powershell: |` needs two more spaces so `displayName` ends up in the right place.

File: eng/pipelines/publish.yml [hunk 3]
```diff
@@ -32,6 +32,18 @@ jobs:
         value: ${{ parameters.artifactsDir }}/AssetManifests
 
     steps:
+      - powershell: |
+          $prefix = "refs/heads/"
```
Comment (safern) [inline eng/pipelines/publish.yml]: Why is this needed? From what I know, Build.SourceBranch uses the full name? i.e -> refs/heads/release/3.0 ?

Comment (dagood) [reply eng/pipelines/publish.yml]: You're right about the var, this describes `Build.SourceBranch` and `Build.SourceBranchName`: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml.

This trimming step is needed because in dotnet/versions, we don't want `refs/heads/release/3.0`, only `release/3.0`.

Note that the manifest still uses `refs/heads/release/3.0` in this PR, if you didn't notice.

Comment (mmitche) [reply eng/pipelines/publish.yml]: Yes that is intentional for the manifest

**Comment by mmitche** [comment]
@safern Already did the update
